stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/message/MessageSendingService.kt

```diff
@@ -42,14 +42,23 @@ internal class MessageSendingService(
                     enrichWithCid(channelController.cid)
                 }
                 user = requireNotNull(domainImpl.user.value)
-                attachments.forEach { attachment ->
+
+                val (attachmentsToUpload, nonFileAttachments) = attachments.partition { it.upload != null }
```

Comment:
Shouldn't we check the `UploadState`? :thinking: 
If the attachment is already uploaded, we shouldn't enqueue it either and be able to send it directly

Reply:
As far as I see, this `MessageSendingService` is just the initial thing that sends messages when someone invokes the `SendMessage` use case through `ChatDomain`. Retries are handled in a different place in the code, they don't come to this service. So this is always the first time we are setting the `attachmentState` for something.